src/utils/general.ts

```diff
@@ -891,11 +893,20 @@ export function waitForPageToBeVisible() {
   });
 }
 
+export function getBrowserCurrentLocale() {
+  try {
+    return chrome.i18n.getUILanguage();
+  } catch (e) {
+    return navigator.language;
+  }
+}
+
 export async function clearCache() {
   const cacheArray = await api.storage.list();
   let deleted = 0;
-
-  j.$.each(cacheArray, function (index, cache) {
+  localStore.clear();
```

Comment:
This is dangerous, because in the userscript it will clear the localStorage of the website. Which is not good. It can not be done in this utility as it can be executed in any context. It should only run in the extension and extension popup

Reply:
I thought when you use userscript and can only press this button from pwa page? Can it still clear website's localStorage too?

Reply:
Malsync used to be a userscript only, without an extra domain and pwa. In the extension we removed the option to do it, but in the userscript you can still load the popup inside of the website. Many people switched to the userscript again just because of that feature.
![image](https://github.com/user-attachments/assets/f859eb38-3bcb-440a-a1a5-c2b7e533b1be)

Reply:
Oh how to you open it? Didn't know, I'll add a check for an extension only

Reply:
As the idea of a util is that it can be used anywhere. I am not that confortable to include it in there.

I just found that we actually clear the localstorage when switching provider. It is probably a better idea to add it to the click event of the clear cache. This makes sure it runs in the context of the popup

https://github.com/MALSync/MALSync/blob/master/src/_minimal/components/settings/settings-structure-tracking.ts#L47